runtime/vm/classloadersearch.c

```diff
@@ -237,3 +242,52 @@ addZipToLoader(J9JavaVM * vm, const char * filename, J9ClassLoader * classLoader
 
 	return rc;
 }
+
+
+/**
+ * Loads the named module using jdk.internal.module.Modules.loadModule()
+ *
+ * @param [in] vm pointer to JVM
+ * @param [in] moduleName name of module to load
+ * @param [out] modulePtr pointer to return module object.  May be null.
+ *
+ * @return CLS_ERROR_NONEE on success, other value on failure
+ */
+UDATA JNICALL
+loadModule(J9JavaVM *vm, const char *moduleName, jobject *modulePtr)
+{
+	UDATA rc = CLS_ERROR_NONE;
+	J9VMThread *currentThread = currentVMThread(vm);
+	JNIEnv *env = (JNIEnv *) currentThread;
+	jclass jimModules = vm->jimModules;
+
+	if (NULL != modulePtr) {
+		*modulePtr = NULL;
+	}
+
+	/*
+	 * vm->jimModules may have been set already by JVMTI.
+	 * If not, grab a private copy of the class.
+	 */
+	if (NULL == jimModules) {
+		jimModules = (*env)->FindClass(env, "jdk/internal/module/Modules");
+	}
+	if (NULL == jimModules) {
+		rc = CLS_ERROR_NOT_FOUND;
+	} else {
+		jmethodID loadModule =
+				(*env)->GetStaticMethodID(env, jimModules, "loadModule", "(Ljava/lang/String;)Ljava/lang/Module;");
+		if (NULL == loadModule) {
+			rc = CLS_ERROR_NOT_FOUND;
+		} else {
+			jstring moduleNameString =  (*env)->NewStringUTF(env, moduleName);
```

Comment:
A check that this succeeded should be added. Also please fix the formatting (extra space after `=`).

Reply:
Thanks.  Fixed.